lib/project_types/script/layers/infrastructure/task_runner.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class TaskRunner
+        TASK_RUNNERS = {
+          "ts" => Infrastructure::AssemblyScriptTaskRunner,
```

Comment:
I think all the language values would be better as constants of `TaskRunner`.  I wonder if it would read a bit more natural if we had something like:
`TaskRunner.for_language(TaskRunner.AssemblyScript, ctx)`. 
Basically, reading it makes it clear that the `TaskRunner` is unique per language.  I also think it would be better if the Script Author is thinking that she is writing in AS, and the compiled type happens to be TS is an infrastructure detail that we can hide.

Reply:
Thats a good idea. I think I'll hold off on that as long as your okay with my approach [here](https://github.com/Shopify/shopify-app-cli/pull/685#discussion_r441696108) because I dont plan on having this file for long - its just the renamed/refactored version of [script_builder.rb](https://github.com/Shopify/shopify-app-cli/pull/685/files#diff-202ae1b73641011a7d8cc143a217cdb5)